### PR TITLE
Dw 3760 secrets manager

### PR DESCRIPTION
**Pull request by TheDogfather17**

### CODE REVIEW

File: aws-api-payload.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
         "StepConcurrencyLevel": 1,
         "TerminationProtected": false,
         "BootstrapActions": [],
-        "LogUri": "s3://$ADG_S3_LOG_BUCKET/emr/aws-ingestion/",
+        "LogUri": "s3://$S3_LOG_BUCKET/emr/aws-ingestion/",
```
Comment (mattburgess) [inline aws-api-payload.json:6]: Should be `s3://$S3_LOG_BUCKET/emr/analytical-dataset-generation/`

File: aws-api-payload.json [hunk 2]
```diff
@@ -136,7 +136,7 @@
         "Configurations": [
             {
                 "Properties": {
-                    "yarn.nodemanager.remote-app-log-dir": "s3://$ADG_S3_LOG_BUCKET/yarn/",
+                    "yarn.nodemanager.remote-app-log-dir": "s3://$S3_LOG_BUCKET/yarn/",
```
Comment (mattburgess) [inline aws-api-payload.json:139]: Should be `s3://$S3_LOG_BUCKET/emr/analytical-dataset-generation/yarn/`

File: aws-api-payload.json [hunk 3]
```diff
@@ -156,7 +156,7 @@
...
                     "spark.driver.extraClassPath": "/usr/lib/hadoop-lzo/lib/*:/usr/lib/hadoop/hadoop-aws.jar:/usr/share/aws/aws-java-sdk/*:/usr/share/aws/emr/emrfs/conf:/usr/share/aws/emr/emrfs/lib/*:/usr/share/aws/emr/emrfs/auxlib/*:/usr/share/aws/emr/goodies/lib/emr-spark-goodies.jar:/usr/share/aws/emr/security/conf:/usr/share/aws/emr/security/lib/*:/usr/share/aws/hmclient/lib/aws-glue-datacatalog-spark-client.jar:/usr/share/java/Hive-JSON-Serde/hive-openx-serde.jar:/usr/share/aws/sagemaker-spark-sdk/lib/sagemaker-spark-sdk.jar:/usr/share/aws/emr/s3select/lib/emr-s3-select-spark-connector.jar:/usr/lib/hive/lib/hive-hbase-handler.jar:/usr/lib/hbase/*:/usr/lib/hive/lib/metrics-core-2.2.0.jar:/usr/lib/hive/lib/htrace-core-3.1.0-incubating.jar",
-                    "spark.sql.warehouse.dir": "s3://$ADG_S3_PUBLISH_BUCKET/data/external",
+                    "spark.sql.warehouse.dir": "s3://$S3_PUBLISH_BUCKET/data/external",
```
Comment (mattburgess) [inline aws-api-payload.json:159]: Should be `s3://$S3_PUBLISH_BUCKET/analytical-dataset/data/external` for now; I reserve the right to change my mind once I see what all of these different outputs actually look like :)

File: aws-api-payload.json [hunk 4]
```diff
@@ -171,7 +171,7 @@
                 "Properties": {
                     "hive.metastore.schema.verification": "false",
                     "hive.metastore.client.factory.class": "com.amazonaws.glue.catalog.metastore.AWSGlueDataCatalogHiveClientFactory",
-                    "hive.metastore.warehouse.dir": "s3://$ADG_S3_PUBLISH_BUCKET/hive/external"
+                    "hive.metastore.warehouse.dir": "s3://$S3_PUBLISH_BUCKET/hive/external"
```
Comment (mattburgess) [inline aws-api-payload.json:174]: Should be `s3://$S3_PUBLISH_BUCKET/analytical-dataset/hive/external`

File: cb-payload.json [hunk 5]
```diff
@@ -17,7 +17,7 @@
             "properties": {
                 "yarn.log-aggregation-enable": "true",
                 "yarn.log-aggregation.retain-seconds": "-1",
-                "yarn.nodemanager.remote-app-log-dir": "s3://$ADG_S3_LOG_BUCKET/yarn/"
+                "yarn.nodemanager.remote-app-log-dir": "s3://$S3_LOG_BUCKET/yarn/"
```
Comment (mattburgess) [inline cb-payload.json:20]: Should be `s3://$S3_LOG_BUCKET/emr/analytical-dataset-generation/yarn/`

File: cb-payload.json [hunk 6]
```diff
@@ -30,7 +30,7 @@
             "classification": "spark-defaults",
             "properties": {
                 "spark.sql.catalogImplementation": "hive",
-                "spark.sql.warehouse.dir": "s3://$ADG_S3_PUBLISH_BUCKET/data/external",
+                "spark.sql.warehouse.dir": "s3://$S3_PUBLISH_BUCKET/data/external",
```
Comment (mattburgess) [inline cb-payload.json:33]: Should be `s3://$S3_LOG_BUCKET/emr/analytical-dataset-generation/yarn/`

File: cb-payload.json [hunk 7]
```diff
@@ -50,13 +50,13 @@
             "properties": {
                 "hive.metastore.client.factory.class": "com.amazonaws.glue.catalog.metastore.AWSGlueDataCatalogHiveClientFactory",
                 "hive.metastore.schema.verification": "false",
-                "hive.metastore.warehouse.dir": "s3://$ADG_S3_PUBLISH_BUCKET/hive/external"
+                "hive.metastore.warehouse.dir": "s3://$S3_PUBLISH_BUCKET/hive/external"
```
Comment (mattburgess) [inline cb-payload.json:53]: Should be `s3://$S3_PUBLISH_BUCKET/analytical-dataset/hive/external`

File: generate-analytical-dataset.py [hunk 8]
```diff
@@ -12,18 +13,21 @@
 from pyspark.sql import Row
 
 
-
 def main():
-    #secrets_manager = boto3.client('secretsmanager')
-    #response = secrets_manager.get_secret_value(SecretId="ADG-Payload")
-    #S3_PUBLISH_BUCKET = response["ADG_S3_PUBLISH_BUCKET"]
-    S3_PUBLISH_BUCKET = "" #Replace with s3 bucket number to test
+    secret_name = "ADG-Payload"
```
[the remark this answers is not in the document]
Comment (TheDogfather17) [reply generate-analytical-dataset.py]: done

File: generate-analytical-dataset.py [hunk 9]
```diff
@@ -12,18 +13,21 @@
...
+    region_name = "eu-west-2"
```
[the remark this answers is not in the document]
Comment (TheDogfather17) [reply generate-analytical-dataset.py]: removed region name as it turned out to be redundant

File: aws-api-payload.json (hunk 1 above)
Comment (TheDogfather17) [reply aws-api-payload.json]: done

File: aws-api-payload.json (hunk 2 above)
Comment (TheDogfather17) [reply aws-api-payload.json]: done

File: aws-api-payload.json (hunk 3 above)
Comment (TheDogfather17) [reply aws-api-payload.json]: done

File: aws-api-payload.json (hunk 4 above)
Comment (TheDogfather17) [reply aws-api-payload.json]: done

**Pull request merged**